packages/next/server/image-optimizer.ts

```diff
@@ -3,6 +3,7 @@ import { createHash } from 'crypto'
 import { createReadStream, promises } from 'fs'
 import { getOrientation, Orientation } from 'get-orientation'
 import { IncomingMessage, ServerResponse } from 'http'
+import micromatch from 'micromatch'
```

Comment:
We also need to validate on the client in `packages/next/client/image.tsx`

I think #18730 is a better implementation because it doesn't rely on a 3rd party package that would need to be bundled in the client.

Reply:
I think that special modules does his job better then that pr. The module has a lot of tests and it works. Pr has unknown status. What if I add `*.staging.*.com` to the config list?

Reply:
In any case we need that feature a lot. That image resizing is super awesome. And there is so many cdn's with random sub domains.

Reply:
> What if I add *.staging.*.com or **.google.com to the config list? 

These are great questions which is why its better to have our own implementation, so we can test and document the usage.

Reply:
Should I continue on this PR - add some more tests and so on?

Reply:
If so, we would need to drop `micromatch`.

I'm skeptical if we even need to differentiate `**` vs `*` but willing to hear out any use cases

Reply:
On second thought, maybe we do need `**` because wildcard certs have a single `*` so we should match that behavior and then `**` could mean any number of subdomains.

We could even introduce this in separate phases so that only `**` is supported at first which can be implemented with a simple `url.hostname.endsWith(suffix)`. Then we can add support for single `*` in a separate PR.

Reply:
@FDiskas @styfle is the decision around support for `**` vs `*` the blocker on merging this PR?

Reply:
The blocker is `micromatch`. It shouldn't use an external dependency since we need to document behavior around `**` vs `*`. See #27925